File: challenges/2026-09-08-mixer-friend-circles/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public static int[] MixerFriendCircles(int n, List<List<int>> introductions)
    {
        // TODO: implement
        return new int[0];
    }

    public static void Main(string[] args)
    {
        var tests = new List<(string Name, int N, List<List<int>> Input, int[] Expected)>
        {
            ("example 1", 5, new List<List<int>> { new List<int>{0,1}, new List<int>{1,2}, new List<int>{3,4} }, new[]{3,0}),
            ("example 2", 6, new List<List<int>> { new List<int>{0,1}, new List<int>{2,3} }, new[]{2,2}),
            ("example 3", 1, new List<List<int>>(), new[]{1,1}),
            ("edge: duplicate introductions", 3, new List<List<int>> { new List<int>{0,1}, new List<int>{1,0}, new List<int>{0,1} }, new[]{2,1}),
            ("edge: full chain merges everyone", 4, new List<List<int>> { new List<int>{0,1}, new List<int>{1,2}, new List<int>{2,3} }, new[]{4,0}),
        };

        foreach (var (name, n, input, expected) in tests)
        {
            int[] got = MixerFriendCircles(n, input);
            string status = got.SequenceEqual(expected) ? "PASS" : "FAIL";
            Console.WriteLine($"[{status}] {name}: expected=[{string.Join(",", expected)}] actual=[{string.Join(",", got)}]");
        }
    }
}

File: challenges/2026-09-08-mixer-friend-circles/solution.go
package main

import (
	"fmt"
	"reflect"
)

func mixerFriendCircles(n int, introductions [][]int) []int {
	// TODO: implement
	return []int{}
}

type testCase struct {
	name     string
	n        int
	input    [][]int
	expected []int
}

func main() {
	tests := []testCase{
		{"example 1", 5, [][]int{{0, 1}, {1, 2}, {3, 4}}, []int{3, 0}},
		{"example 2", 6, [][]int{{0, 1}, {2, 3}}, []int{2, 2}},
		{"example 3", 1, [][]int{}, []int{1, 1}},
		{"edge: duplicate introductions", 3, [][]int{{0, 1}, {1, 0}, {0, 1}}, []int{2, 1}},
		{"edge: full chain merges everyone", 4, [][]int{{0, 1}, {1, 2}, {2, 3}}, []int{4, 0}},
	}

	for _, tc := range tests {
		got := mixerFriendCircles(tc.n, tc.input)
		status := "FAIL"
		if reflect.DeepEqual(got, tc.expected) {
			status = "PASS"
		}
		fmt.Printf("[%s] %s: expected=%v actual=%v\n", status, tc.name, tc.expected, got)
	}
}
